src/plugins/vis_type_vega/public/expressions/helpers.ts

```diff
@@ -49,7 +50,22 @@ export const formatDatatable = (
   datatable.columns.forEach((column) => {
     // clean quotation marks from names in columns
     column.name = cleanString(column.name);
+    // clean ids to remove "." as that will cause vega to not process it correctly.
```

Comment:
just a question here: can we just rename the ids so they do not include '.'?

Reply:
If we change it how it is in `OpenSearchDashboardsDatatable`, this can cause a lot of implications as we need to modify `OpenSearchaggsExpressionFunction`.